ddtrace/tracer.py

```diff
@@ -292,19 +293,24 @@ def configure(
                 default_hostname = agent.get_hostname()
                 default_port = agent.get_trace_port()
 
-            if hasattr(self, "writer") and self.writer.is_alive():
+            if hasattr(self, "writer") and self.writer and self.writer.is_alive():
                 self.writer.stop()
 
-            self.writer = AgentWriter(
-                hostname or default_hostname,
-                port or default_port,
-                uds_path=uds_path,
-                https=https,
-                sampler=self.sampler,
-                priority_sampler=self.priority_sampler,
-                dogstatsd=get_dogstatsd_client(self._dogstatsd_url),
-                report_metrics=config.health_metrics_enabled,
-            )
+            self._agent_hostname = hostname or default_hostname
+            self._agent_port = port or default_port
+            self._agent_uds_path = uds_path
+            self._agent_https = https
+            self._agent_started = False
+
+            try:
+                uwsgi.check_uwsgi(self._init_agent_writer, atexit=self.shutdown)
+            except uwsgi.uWSGIMasterProcess:
+                self.writer = None
```

Comment:
We gotta be sure here that all places where a writer is referenced are safe-guarded with a null check

Reply:
Added these checks